consumer/pdata/trace.go

```diff
@@ -40,15 +40,25 @@ func TracesToOtlp(td Traces) []*otlptrace.ResourceSpans {
 	return *td.orig
 }
 
-// ToOtlpProtoBytes returns the internal Traces to OTLP Collector
-// ExportTraceServiceRequest ProtoBuf bytes. This is intended to export OTLP
-// Protobuf bytes for OTLP/HTTP transports.
+// ToOtlpProtoBytes converts the internal Traces to OTLP Collector
+// ExportTraceServiceRequest ProtoBuf bytes.
 func (td Traces) ToOtlpProtoBytes() ([]byte, error) {
 	return proto.Marshal(&otlpcollectortrace.ExportTraceServiceRequest{
 		ResourceSpans: *td.orig,
 	})
 }
 
+// TracesFromOtlpProtoBytes converts OTLP Collector ExportTraceServiceRequest
+// ProtoBuf bytes to the internal Traces.
+func TracesFromOtlpProtoBytes(data []byte) (Traces, error) {
```

Comment:
Can we make this a function on the Traces struct? That way it will be consistent with gogoproto, and will allow reusing of the objects if possible.

Reply:
if you do that does it imply a merge / append or what?

Reply:
Overrides the entire thing. No merge, no append.

Reply:
I like it as a function on the struct. It might be less obvious about merge / append thing, but to my mind consistency and avoiding allocations worth it.